src/include/duckdb/main/capi/capi_internal_table.hpp

```diff
@@ -0,0 +1,77 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/main/capi/capi_internal.hpp
```

Comment:
Align with file name. Maybe we can call this `capi_table_functions.hpp`? If we're introducing more header files for the C API implementation side, then let's give them more descriptive names, and eventually call `capi_internal` something like `capi_common` or so?